### PR TITLE
Edit handle scale improvements

### PR DESCRIPTION
**Pull request by dback2**
Replace the proportional scaling cubes at the corners of the bounding box with a single scale cube in the center of the entity. This cube behaves like the scale cube in Unity in that rightward or upward mouse movement will scale outward, while leftward or downward mouse movement will scale inward. As well, once scaling downward passed the minimum dimensions, it will invert and start scaling back upward.

Replace the single axis stretching spheres with cubes.

Maintain stretch cube positioning relative to mouse when stretching, then revert back to default handle positioning when done stretching.

Addresses:
https://highfidelity.manuscript.com/f/cases/17467
https://highfidelity.manuscript.com/f/cases/17474
https://highfidelity.manuscript.com/f/cases/17470

Test Plan:
- Enter Interface in desktop mode
- In Create app, create any entity and select it or select any existing entity
- Verify the edit handles to stretch each axis are now cubes instead of spheres
- Verify that when dragging a stretch cube on each axis that the stretch cube stays under the mouse's position until it is released, at which point it goes back to it's default position aligned with the other handles
- Verify the edit handle cubes at the corners of the bounding box are now gone
- Verify the new edit handle cube at the center of the entity performs proportional scaling when dragged:
-- rightward or upward mouse movement proportionally scales the entity upward
-- leftward or downward mouse movement will scale it downward
-- scaling downward passed minimum dimensions will invert and start scaling upward again
- Verify the center scale cube proportionally scales entities with dimensions of (100,1,1), (1,100,1), and (1,1,100) all at the same rate

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12325/HighFidelity-Beta-PR13865-8c422cb.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12326/HighFidelity-Beta-PR13865-8eddb36.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12326/HighFidelity-Beta-PR13865-8eddb36.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12326/HighFidelity-Beta-Interface-PR13865-8eddb36.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12326/HighFidelity-Beta-PR13865-8eddb36.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12326/HighFidelity-Beta-Interface-PR13865-8eddb36.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12393/HighFidelity-Beta-PR13865-30c6975.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12393/HighFidelity-Beta-PR13865-30c6975.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12393/HighFidelity-Beta-Interface-PR13865-30c6975.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12393/HighFidelity-Beta-PR13865-30c6975.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12393/HighFidelity-Beta-Interface-PR13865-30c6975.exe

**Comment by alexiamandeville** [comment]
Tested. My feedback:
Scales too fast. Reduce by 25%?
Scaling in the negative direction should inverse the object once it hits the 0 point.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12720/HighFidelity-Beta-PR13865-6ac3819.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12720/HighFidelity-Beta-PR13865-6ac3819.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12720/HighFidelity-Beta-Interface-PR13865-6ac3819.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12720/HighFidelity-Beta-PR13865-6ac3819.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12720/HighFidelity-Beta-Interface-PR13865-6ac3819.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12743/HighFidelity-Beta-PR13865-3d75d47.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12743/HighFidelity-Beta-PR13865-3d75d47.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12743/HighFidelity-Beta-Interface-PR13865-3d75d47.dmg

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
New handles are cubes and are very responsive.

File: scripts/system/libraries/entitySelectionTool.js [hunk 1]
```diff
@@ -1424,127 +1394,56 @@ SelectionDisplay = (function() {
...
+            RBFPosition = Vec3.sum(position, Vec3.multiplyQbyV(rotation, RBFPosition));
+            var LTNPosition = { x: -edgeOffsetX, y: edgeOffsetY, z: -edgeOffsetZ };
+            LTNPosition = Vec3.sum(position, Vec3.multiplyQbyV(rotation, LTNPosition));
+            var RTNPosition = { x: edgeOffsetX, y: edgeOffsetY, z: -edgeOffsetZ };
+            RTNPosition = Vec3.sum(position, Vec3.multiplyQbyV(rotation, RTNPosition));
+            var LTFPosition = { x: -edgeOffsetX, y: edgeOffsetY, z: edgeOffsetZ };
+            LTFPosition = Vec3.sum(position, Vec3.multiplyQbyV(rotation, LTFPosition));
+            var RTFPosition = { x: edgeOffsetX, y: edgeOffsetY, z: edgeOffsetZ };
+            RTFPosition = Vec3.sum(position, Vec3.multiplyQbyV(rotation, RTFPosition));
+            Overlays.editOverlay(handleScaleTREdge, { start: RTNPosition, end: RTFPosition });
```
Comment (huffman) [inline scripts/system/libraries/entitySelectionTool.js]: Do we still use these?

Comment (dback2) [reply scripts/system/libraries/entitySelectionTool.js]: These edges are the bounding box edge overlays, they are probably not well-named anymore as they were connected with the scale cubes before. Should we rename?

File: scripts/system/libraries/entitySelectionTool.js [hunk 2]
```diff
@@ -2351,34 +2282,26 @@ SelectionDisplay = (function() {
...
-                    percentChange = changeInDimensions.z / initialDimensions.z;
-                }
+                var viewportDimensions = Controller.getViewportDimensions();
+                var mouseXDifference = (event.x - beginMouseEvent.x) / viewportDimensions.x;
+                var mouseYDifference = (beginMouseEvent.y - event.y) / viewportDimensions.y;
+                var mouseDifference = mouseXDifference + mouseYDifference;
+                var toCameraDistance = getDistanceToCamera(position);   
+                var dimensionsMultiple = toCameraDistance * STRETCH_DIRECTION_ALL_CAMERA_DISTANCE_MULTIPLE; 
+                var dimensionChange = mouseDifference * dimensionsMultiple;
+                percentChange = dimensionChange / initialDimensions.z;
```
Comment (huffman) [inline scripts/system/libraries/entitySelectionTool.js]: I wonder if there isn't a more predictable divisor here - maybe the average of all dimensions? Scaling a 1x1x500 cube will be different from a 500x1x1, right?

Comment (dback2) [reply scripts/system/libraries/entitySelectionTool.js]: You're right this should probably be changed to average, I was copying the case of all changeOfDimensions axes being equal before. Cases like you mention do make a difference yeah.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12833/HighFidelity-Beta-PR13865-1ad3a09.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12833/HighFidelity-Beta-Interface-PR13865-1ad3a09.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12833/HighFidelity-Beta-PR13865-1ad3a09.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12833/HighFidelity-Beta-Interface-PR13865-1ad3a09.exe

**Comment by QliCknox** [comment]
Currently Testing

**Comment by QliCknox** [comment]
Fail @conklin94122 

Entity does not scale inversely when reduced past '0'
[Archive.zip](https://github.com/highfidelity/hifi/files/2371408/Archive.zip)

**Comment by dback2** [comment]
@QliCknox Scaling inversely is only supported on the center scale cube for proportional scaling, other than that did everything else check out?

**Comment by QliCknox** [comment]
PASS @conklin94122 

Retested this PR after communication with @dback2 
The cube does scale inversely when reduced past 0

**Pull request merged**